### ISSUE TITLE
[v2.2] Postgres role 'nobody' can't create submissions (permission denied for relation patchwork_submission)

### ISSUE THREAD

**Issue by alialnu** [getpatchwork/patchwork#364]
The role `nobody` seems to no longer have sufficient permissions to create a submission since v2.2:

```
$ cat mbox | sudo -u nobody -- /path/to/patchwork/patchwork/bin/parsemail.sh

Error when parsing incoming email: ProgrammingError('permission denied for relation patchwork_submission\n',)
Traceback (most recent call last):
  File "/path/to/virtualenv/lib/python3.5/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
psycopg2.errors.InsufficientPrivilege: permission denied for relation patchwork_submission


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/path/to/patchwork/patchwork/management/commands/parsemail.py", line 66, in handle
    result = parse_mail(mail, options['list_id'])
  File "/path/to/patchwork/patchwork/parser.py", line 1210, in parse_mail
    series.add_patch(patch, x)
  File "/path/to/patchwork/patchwork/models.py", line 768, in add_patch
    patch.save()
  File "/path/to/patchwork/patchwork/models.py", line 499, in save
    super(Patch, self).save(**kwargs)
  File "/path/to/patchwork/patchwork/models.py", line 353, in save
    super(EmailMixin, self).save(*args, **kwargs)
  File "/path/to/virtualenv/lib/python3.5/site-packages/django/db/models/base.py", line 741, in save
    force_update=force_update, update_fields=update_fields)
  File "/path/to/virtualenv/lib/python3.5/site-packages/django/db/models/base.py", line 776, in save_base
    parent_inserted = self._save_parents(cls, using, update_fields)
  File "/path/to/virtualenv/lib/python3.5/site-packages/django/db/models/base.py", line 807, in _save_parents
    force_insert=parent_inserted,
  File "/path/to/virtualenv/lib/python3.5/site-packages/django/db/models/base.py", line 851, in _save_table
    forced_update)
  File "/path/to/virtualenv/lib/python3.5/site-packages/django/db/models/base.py", line 900, in _do_update
    return filtered._update(values) > 0
  File "/path/to/virtualenv/lib/python3.5/site-packages/django/db/models/query.py", line 760, in _update
    return query.get_compiler(self.db).execute_sql(CURSOR)
  File "/path/to/virtualenv/lib/python3.5/site-packages/django/db/models/sql/compiler.py", line 1469, in execute_sql
    cursor = super().execute_sql(result_type)
  File "/path/to/virtualenv/lib/python3.5/site-packages/django/db/models/sql/compiler.py", line 1140, in execute_sql
    cursor.execute(sql, params)
  File "/path/to/virtualenv/lib/python3.5/site-packages/django/db/backends/utils.py", line 99, in execute
    return super().execute(sql, params)
  File "/path/to/virtualenv/lib/python3.5/site-packages/django/db/backends/utils.py", line 67, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
  File "/path/to/virtualenv/lib/python3.5/site-packages/django/db/backends/utils.py", line 76, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/path/to/virtualenv/lib/python3.5/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
  File "/path/to/virtualenv/lib/python3.5/site-packages/django/db/utils.py", line 89, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/path/to/virtualenv/lib/python3.5/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
django.db.utils.ProgrammingError: permission denied for relation patchwork_submission
```

My DB driver is Postgres, and the role `nobody` has access privileges `INSERT` and `SELECT` on table `patchwork_submission` (given by [grant-all.postgres.sql](https://github.com/getpatchwork/patchwork/blob/master/lib/sql/grant-all.postgres.sql)).

Would you agree with making the following change as a fix?:
```
diff --git a/lib/sql/grant-all.postgres.sql b/lib/sql/grant-all.postgres.sql
index 56a2486..427c3e8 100644
--- a/lib/sql/grant-all.postgres.sql
+++ b/lib/sql/grant-all.postgres.sql
@@ -72,7 +72,9 @@ GRANT INSERT, SELECT ON
        patchwork_comment,
        patchwork_coverletter,
        patchwork_event,
-       patchwork_seriesreference,
+       patchwork_seriesreference
+TO "nobody";
+GRANT INSERT, SELECT, UPDATE ON
        patchwork_submission
 TO "nobody";
 GRANT INSERT, SELECT, UPDATE, DELETE ON
```

**Comment by jkloetzke** [comment]
I can confirm the problem on a fresh install of 2.2.2 (on Debian Buster if it matters). The patch above solves the problem. :confetti_ball:  Thanks!

**Comment by stephenfin** [comment]
Resolves on stable/2.2. Thanks for the fix!

**Comment by alialnu** [comment]
> 
> 
> Resolved on stable/2.2. Thanks for the fix!

Thanks @stephenfin
